1. Consider *n* people who are attending a party. We assume that every person has
   an equal probability of being born on any day during the year, independent of
   everyone else, and ignore leap years (i.e. year has 365 days). What is the
   probability that each person has a distinct birthday? How many people are
   necessary to obtain a 99% probability of at least one birthday collision (i.e. 2 people
   with the same birthday).

In [73]:
def birthday(n):
    '''
    Find the probablity of matching birthdays in a group of n people
    '''
    # Probablity of 2 people matching
    prob_match = (364/365)
    
    #total combinations of people allowing for more chance to match
    total_combos = (n*(n-1))/2
    
    # Probablity of getting two birthdays being the same
    prob_total = prob_match ** total_combos
    
    return 1- prob_total

    
    

In [86]:
def nine_nine_search():
    '''
    Search through a set of n till you find at what threshhold is 99% chance
    '''
    
    prob_match = (364/365)
    
    for i in range(1000):
        total_combos = (i*(i-1))/2
        prob_total = prob_match**total_combos
        
        if (1-prob_total) >=0.9855:
            return i
            
        

In [87]:
nine_nine_search()

57

In [67]:
birthday(57)

1596.0


0.9874571383670745

2. A hunter has two hunting dogs. One day, on the trail of some animal, the
   hunter comes to a place where the road diverges into two paths. He knows that
   each dog, independent of the other, will choose the correct path with
   probability *p*. The hunter decides to let each dog choose a path, and if they
   agree, he takes that one, and if they disagree, he randomly picks a path. Is his
   strategy better than just letting one of the two dogs decide on a path?
   Explain why or why not.

In [65]:
# Assume dogs are better than random choice

def path_choice(dog1 = 0.8, dog2 = 0.7):
    agree = dog1*dog2
    disagree  = 1 - agree
    
    #P(R | Disagree) = P(Disagree|R)*P(R)
    #P(R) = 0.5
    # P(Disaggree|R) *0.42
    
    bayes_choice = (0.42*0.5)/(0.6*0.5 + 0.7*0.5)
    return agree, bayes_choice


In [66]:
path_choice()

(0.5599999999999999, 0.3230769230769231)

The probablity that both dogs choose the same path and are correct is 0.56 but the probablity of choosing the correct path given that they didn't agree would be 0.5. As long as the probablity of each individual dog is greater than 0.5 random you should choose a random dog. For the dogs agreeing to work they would need to have an average of 0.75 probablity of deciding correctly to be better than random. 